src/components/CacheFS.brs

```diff
@@ -0,0 +1,58 @@
+' @import /components/rokuComponents/EVPDigest.brs
+function CacheFS() as Object
+  prototype = {}
+
+  prototype._PREFIX = "cachefs:/"
+
+  prototype._digest = Invalid
+
+  _constructor = function (m as Object) as Object
+    m._digest = EVPDigest()
+    m._digest.setup("sha1")
+
+    return m
+  end function
+
+  ' Reads data stored in cachefs under the specific key
+  ' @param {String} key
+  ' @returns {Object}
+  prototype.read = function (key as String) as Object
+    if (key = "") then return Invalid
+
+    content = ReadAsciiFile(m._PREFIX + m._hash(key))
+    if (content = "") then return Invalid
```

Comment:
The empty string is valid JSON response so it should be respected. I would remove it and let empty string to be returned.

Reply:
The problem is that ReadAsciiFile will return an empty string for a non-existing file too, so by removing this check, we wouldn't know if data under the given key exist

Reply:
Not sure if this is applicable in this case but maybe you could check if file exists using `Exists` https://developer.roku.com/en-gb/docs/references/brightscript/interfaces/iffilesystem.md#existspath-as-string-as-boolean?